lua/scriptwrapper.lua

```diff
@@ -26,27 +28,62 @@ local TryRunSandboxed = function(fn)
 	end
 end
 
+local function merge(dest, source)
+	for k,v in pairs(source) do
+		dest[k] = dest[k] or v
+	end
+	return dest
+end
+
+local TryRun = function(fn)
+	local env = merge(environment, _G)
+	setfenv(fn, env)
+	local success, err = xpcall(fn, PrintStackTrace)
+	if not success then
+		FatalError(err)
+	end
+end
+
 WorldLoaded = function()
 	if environment.WorldLoaded ~= nil then
-		TryRunSandboxed(environment.WorldLoaded)
+		if isSandboxed then
+			TryRunSandboxed(environment.WorldLoaded)
+		else
+			TryRun(environment.WorldLoaded)
+		end
 	end
 end
 
 Tick = function()
 	if environment.Tick ~= nil then
-		TryRunSandboxed(environment.Tick)
+		if isSandboxed then
+			TryRunSandboxed(environment.Tick)
+		else
+			TryRun(environment.Tick)
+		end
 	end
 end
 
 ExecuteSandboxedScript = function(file, contents)
 	local script, err = loadstring(contents, file)
+	isSandboxed = true
 	if (script == nil) then
 		FatalError("Error parsing " .. file .. ". Reason: " .. err)
 	else
 		TryRunSandboxed(script)
 	end
 end
 
+ExecuteScript = function(file, contents)
```

Comment:
We're introducing a lot of duplication into this file that i'm not sure is necessary.

Can we instead have a single `ExecuteScript` function that takes `isSandboxed` as an argument, and then a single `TryRun` that looks something like

```lua
local TryRun = function(fn)
        if isSandboxed then
                origfn = fn
                fn = function() sandbox.run(origfn, {env = environment, quota = MaxUserScriptInstructions}) end
        else
                -- TODO: Explain what this is doing
                local env = merge(environment, _G)
                setfenv(fn, env)
        end

	local success, err = xpcall(fn, PrintStackTrace)
	if not success then
		FatalError(err)
	end
end
```
?

Reply:
You can't set `isSandboxed` from outside so it will always stay false.